bwi_virtour: compile dependency error

```
/home/travis/catkin_ws/src/bwi_common/bwi_virtour/src/go_to_location_service.cpp:7:32: fatal error: bwi_virtour/Rotate.h: No such file or directory
 #include "bwi_virtour/Rotate.h"
                                ^
compilation terminated.
```

This looks like an ``add_dependencies()`` error. We've been seeing a lot of those.

xref: #63

--- a/bwi_virtour/CMakeLists.txt
+++ b/bwi_virtour/CMakeLists.txt
@@ -1,58 +1,13 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(bwi_virtour)
 
-## Find catkin macros and libraries
-## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
-## is used, also find other catkin packages
 find_package(catkin REQUIRED COMPONENTS 
-  std_msgs
   actionlib
   roscpp
   message_generation
   bwi_kr_execution
 )
 
-## System dependencies are found with CMake's conventions
-# find_package(Boost REQUIRED COMPONENTS system)
-
-
-## Uncomment this if the package has a setup.py. This macro ensures
-## modules and global scripts declared therein get installed
-## See http://ros.org/doc/api/catkin/html/user_guide/setup_dot_py.html
-# catkin_python_setup()
-
-################################################
-## Declare ROS messages, services and actions ##
-################################################
-
-## To declare and build messages, services or actions from within this
-## package, follow these steps:
-## * Let MSG_DEP_SET be the set of packages whose message types you use in
-##   your messages/services/actions (e.g. std_msgs, actionlib_msgs, ...).
-## * In the file package.xml:
-##   * add a build_depend and a run_depend tag for each package in MSG_DEP_SET
-##   * If MSG_DEP_SET isn't empty the following dependencies might have been
-##     pulled in transitively but can be declared for certainty nonetheless:
-##     * add a build_depend tag for "message_generation"
-##     * add a run_depend tag for "message_runtime"
-## * In this file (CMakeLists.txt):
-##   * add "message_generation" and every package in MSG_DEP_SET to
-##     find_package(catkin REQUIRED COMPONENTS ...)
-##   * add "message_runtime" and every package in MSG_DEP_SET to
-##     catkin_package(CATKIN_DEPENDS ...)
-##   * uncomment the add_*_files sections below as needed
-##     and list every .msg/.srv/.action file to be processed
-##   * uncomment the generate_messages entry below
-##   * add every package in MSG_DEP_SET to generate_messages(DEPENDENCIES ...)
-
-## Generate messages in the 'msg' folder
-# add_message_files(
-#   FILES
-#   Message1.msg
-#   Message2.msg
-# )
-
-## Generate services in the 'srv' folder
 add_service_files(
   FILES
   GoToLocation.srv
@@ -65,119 +20,26 @@ add_service_files(
   Authenticate.srv
 )
 
-## Generate actions in the 'action' folder
-# add_action_files(
-#   FILES
-#   Action1.action
-#   Action2.action
-# )
-
-## Generate added messages and services with any dependencies listed here
 generate_messages(
   DEPENDENCIES
   std_msgs
 )
-
-###################################
-## catkin specific configuration ##
-###################################
-## The catkin_package macro generates cmake config files for your package
-## Declare things to be passed to dependent projects
-## INCLUDE_DIRS: uncomment this if you package contains header files
-## LIBRARIES: libraries you create in this project that dependent projects also need
-## CATKIN_DEPENDS: catkin_packages dependent projects also need
-## DEPENDS: system dependencies of this project that dependent projects also need
 catkin_package(
-#  INCLUDE_DIRS include
-#  LIBRARIES bwi_virtour
-#  CATKIN_DEPENDS other_catkin_pkg
-#  DEPENDS system_lib
+  CATKIN_DEPENDS roscpp actionlib message_generation bwi_kr_execution
 )
 
-###########
-## Build ##
-###########
-
-## Specify additional locations of header files
-## Your package locations should be listed before other locations
-# include_directories(include)
 include_directories(
   ${catkin_INCLUDE_DIRS}
 )
 
-## Declare a cpp library
-# add_library(bwi_virtour
-#   src/${PROJECT_NAME}/bwi_virtour.cpp
-# )
-
 add_executable(go_to_location_service_node src/go_to_location_service.cpp)
-add_dependencies(go_to_location_service_node ${catkin_EXPORTED_TARGETS})
+add_dependencies(go_to_location_service_node ${catkin_EXPORTED_TARGETS} ${${PROJECT_NAME}_EXPORTED_TARGETS})
 target_link_libraries(go_to_location_service_node ${catkin_LIBRARIES})
 
 add_executable(rotate_service_node src/rotateService.cpp)
-add_dependencies(rotate_service_node ${catkin_EXPORTED_TARGETS})
+add_dependencies(rotate_service_node ${catkin_EXPORTED_TARGETS} ${${PROJECT_NAME}_EXPORTED_TARGETS})
 target_link_libraries(rotate_service_node ${catkin_LIBRARIES})
 
 add_executable(tour_manager src/tour_manager.cpp)
-add_dependencies(tour_manager ${catkin_EXPORTED_TARGETS})
+add_dependencies(tour_manager ${catkin_EXPORTED_TARGETS} ${${PROJECT_NAME}_EXPORTED_TARGETS})
 target_link_libraries(tour_manager ${catkin_LIBRARIES})
-
-## Declare a cpp executable
-# add_executable(bwi_virtour_node src/bwi_virtour_node.cpp)
-
-## Add cmake target dependencies of the executable/library
-## as an example, message headers may need to be generated before nodes
-# add_dependencies(bwi_virtour_node bwi_virtour_generate_messages_cpp)
-
-## Specify libraries to link a library or executable target against
-# target_link_libraries(bwi_virtour_node
-#   ${catkin_LIBRARIES}
-# )
-
-#############
-## Install ##
-#############
-
-# all install targets should use catkin DESTINATION variables
-# See http://ros.org/doc/api/catkin/html/adv_user_guide/variables.html
-
-## Mark executable scripts (Python etc.) for installation
-## in contrast to setup.py, you can choose the destination
-# install(PROGRAMS
-#   scripts/my_python_script
-#   DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-# )
-
-## Mark executables and/or libraries for installation
-# install(TARGETS bwi_virtour bwi_virtour_node
-#   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-#   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-#   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-# )
-
-## Mark cpp header files for installation
-# install(DIRECTORY include/${PROJECT_NAME}/
-#   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-#   FILES_MATCHING PATTERN "*.h"
-#   PATTERN ".svn" EXCLUDE
-# )
-
-## Mark other files for installation (e.g. launch and bag files, etc.)
-# install(FILES
-#   # myfile1
-#   # myfile2
-#   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-# )
-
-#############
-## Testing ##
-#############
-
-## Add gtest based cpp test target and link libraries
-# catkin_add_gtest(${PROJECT_NAME}-test test/test_bwi_virtour.cpp)
-# if(TARGET ${PROJECT_NAME}-test)
-#   target_link_libraries(${PROJECT_NAME}-test ${PROJECT_NAME})
-# endif()
-
-## Add folders to be run by python nosetests
-# catkin_add_nosetests(test)

--- a/bwi_virtour/package.xml
+++ b/bwi_virtour/package.xml
@@ -15,16 +15,22 @@
 
 
   <!-- Url tags are optional, but mutiple are allowed, one per tag -->
-  <url type="repository">https://github.com/utexas-bwi/bwi_experimental/bwi_virtour</url>
-  <url type="bugtracker">https://github.com/utexas-bwi/bwi_experimental/issues</url>
+  <url type="repository">https://github.com/utexas-bwi/bwi_common/bwi_virtour</url>
+  <url type="bugtracker">https://github.com/utexas-bwi/bwi_common/issues</url>
 
   <!-- The *_depend tags are used to specify dependencies -->
   <build_depend>message_generation</build_depend>
   <build_depend>bwi_kr_execution</build_depend>
+  <build_depend>roscpp</build_depend>
+  <build_depend>actionlib</build_depend>
+
   <run_depend>bwi_kr_execution</run_depend>
   <run_depend>message_runtime</run_depend>
+  <run_depend>message_generation</run_depend>
   <run_depend>rosbridge_server</run_depend>
   <run_depend>web_video_server</run_depend>
+  <run_depend>roscpp</run_depend>
+  <run_depend>actionlib</run_depend>
   <buildtool_depend>catkin</buildtool_depend>
 
 

--- a/bwi_virtour/src/rotateService.cpp
+++ b/bwi_virtour/src/rotateService.cpp
@@ -49,9 +49,6 @@ int main(int argc, char**argv) {
   ros::init(argc, argv, "rotate_service_node");
   ros::NodeHandle n;
 
-  //client = new Client("/action_executor/execute_plan", true);
-
-  //ros::Publisher cmd_vel_pub = n.advertise<std_msgs::String>("/cmd_vel", 1000);
   cmd_vel_pub = n.advertise<geometry_msgs::Twist>("/cmd_vel", 1000);
   ros::ServiceServer service = n.advertiseService("rotate", rotateRequest);
   ROS_INFO("Rotate Service Started");

